tools/testing/target_determination/heuristics/interface.py

```diff
@@ -154,6 +162,131 @@ def _print_tests(label: str, tests: StringTuple) -> None:
+
+    def get_priority_info_for_test(self, test_name: str) -> Dict[str, Any]:
+        """Given a failing test, returns information about it's prioritization that we want to emit in our metrics."""
+        return {
+            METRIC_RELEVANCE_GROUP: self._get_test_relevance_group(test_name).name,
+            METRIC_ORDER_WITHIN_RELEVANCE_GROUP: self._get_test_order_within_relevance_group(
+                test_name
+            ),
+            METRIC_NUM_TESTS_IN_RELEVANCE_GROUP: len(
+                self._test_priorities[self._get_test_relevance_group(test_name).value]
+            ),
+            METRIC_ORDER_OVERALL: self._get_test_order(test_name),
+        }
+
+
+class AggregatedHeuristics:
+    """
+    Aggregates the results across all heuristics.
+
+    It saves the individual results from each heuristic and exposes an aggregated view.
+    """
+
+    _heuristic_results: Dict[
+        str, TestPrioritizations
+    ]  # Key is the Heuristic's name. Dicts will preserve the order of insertion, which is important for sharding
+
+    unranked_tests: Tuple[str, ...]
+
+    def __init__(self, unranked_tests: List[str]) -> None:
+        self.unranked_tests = tuple(unranked_tests)
+        self._heuristic_results = {}
+
+    def add_heuristic_results(
+        self, heuristic_name: str, heuristic_results: TestPrioritizations
+    ) -> None:
+        if heuristic_name in self._heuristic_results:
+            raise ValueError(f"We already have heuristics for {heuristic_name}")
+
+        self._heuristic_results[heuristic_name] = heuristic_results
+
+    def get_aggregated_priorities(self) -> TestPrioritizations:
+        """
+        Returns the aggregated priorities across all heuristics.
+        """
+        aggregated_priorities = TestPrioritizations(
+            tests_being_ranked=self.unranked_tests
+        )
+
+        for heuristic_results in self._heuristic_results.values():
+            aggregated_priorities.integrate_priorities(heuristic_results)
+
+        return aggregated_priorities
+
+    def get_test_stats(self, test: str) -> Dict[str, Any]:
+        """
+        Returns the aggregated statistics for a given test.
+        """
+        stats: Dict[str, Any] = {
+            "test_name": test,
+        }
+        heuristics = []
+
+        # Figure out which heuristic gave this test the highest priority
+        highest_ranking_heuristic = None
+        highest_ranking_heuristic_order: int = sys.maxsize
+
+        # And figure out how many heuristics suggested prioritizing this test
+        num_heuristics_prioritized_by = 0
+
+        for heuristic_name, heuristic_results in self._heuristic_results.items():
+            metrics = heuristic_results.get_priority_info_for_test(test)
+            metrics["heuristic_name"] = heuristic_name
+            heuristics.append(metrics)
+
+            if metrics[METRIC_ORDER_OVERALL] > highest_ranking_heuristic_order:
```

Comment:
Should this be `if highest_ranking_heuristic or metrics[METRIC_ORDER_OVERALL] > highest_ranking_heuristic_order:` as `highest_ranking_heuristic_order` is `sys.maxsize` initially?